habitat/tasks/nav/multi_object_nav_task.py

```diff
@@ -0,0 +1,551 @@
+
+    def _get_uuid(self, *args: Any, **kwargs: Any):
+        return self.cls_uuid
+
+    def reset_metric(self, *args: Any, episode, task, **kwargs: Any):
+        task.measurements.check_measure_dependencies(
+            self.uuid, [CurrentGoalSuccess.cls_uuid]
+        )
+        self.update_metric(*args, episode=episode, task=task, **kwargs)
+
+    def update_metric(
+        self,
+        *args: Any,
+        episode: MultiObjectGoalNavEpisode,
+        task: MultiObjectNavigationTask,
+        **kwargs: Any,
+    ):
+        current_goal_success = task.measurements.measures[
+            CurrentGoalSuccess.cls_uuid
+        ].get_metric()
+
+        if current_goal_success == 1 and task.current_goal_index == len(
+            episode.goals
+        ):
+            self._metric = 1
+        else:
+            self._metric = 0
+
+
+@registry.register_measure
+class MultiONSPL(Measure):
+    r"""SPL (Success weighted by Path Length) for sequential goals.
+
+    ref: MultiON: Benchmarking Semantic Map Memory using Multi-Object
+    Navigation - Wani et. al
+    https://arxiv.org/abs/2012.03912
+    The measure depends on MultiONSuccess measure.
+    """
+
+    def __init__(
+        self, sim: Simulator, config: Config, *args: Any, **kwargs: Any
+    ):
+        self._agent_episode_distance: Optional[float] = None
+        self._sim = sim
+        self._config = config
+
+        super().__init__()
+
+    def _get_uuid(self, *args: Any, **kwargs: Any) -> str:
+        return "multiON_spl"
+
+    def reset_metric(
+        self,
+        episode: MultiObjectGoalNavEpisode,
+        task,
+        *args: Any,
+        **kwargs: Any,
+    ):
+        task.measurements.check_measure_dependencies(
+            self.uuid, [MultiONSuccess.cls_uuid]
+        )
+
+        self._agent_episode_distance = 0.0
+        self._previous_position = self._sim.get_agent_state().position
+        self._start_end_episode_distance = episode.info["geodesic_distance"]
+        self.update_metric(*args, episode=episode, task=task, **kwargs)
+
+    def _euclidean_distance(self, position_a, position_b):
+        return np.linalg.norm(position_b - position_a, ord=2)
+
+    def update_metric(
+        self,
+        *args: Any,
+        episode,
+        task: MultiObjectNavigationTask,
+        **kwargs: Any,
+    ):
+        ep_success = task.measurements.measures[
+            MultiONSuccess.cls_uuid
+        ].get_metric()
+
+        current_position = self._sim.get_agent_state().position
+        self._agent_episode_distance += self._euclidean_distance(
+            current_position, self._previous_position
+        )
+
+        self._previous_position = current_position
+
+        self._metric = ep_success * (
+            self._start_end_episode_distance
+            / max(
+                self._start_end_episode_distance, self._agent_episode_distance
+            )
+        )
+
+
+@registry.register_measure
+class MultiONPPL(Measure):
```

Comment:
@mathfac @Skylion007 what are you thoughts on generalizing this?  This could be defined for other navigation tasks as
```python
progress * (progress * init_dist_to_goal / max(agent_travel, progress * init_dist_to_goal))
```

This is basically SoftSPL but the inverse Path Length term is weight by the length of the shortest path to reach current progress instead of always shortest path to goal and is actually something that we've discussed using instead.